dsa-23au/java-dsa/NMac-app/src/Dining_Philosophers.java

```diff
@@ -0,0 +1,111 @@
+// Chapter 15: Threads and Locks
+// pp. 174-179. 15.3
+
+
+// Dining Philosophers: In the famous dining philosopher’s problem,
+// a bunch of philosophers are sitting around a circular table with one chopstick between each of them.
+// A philosopher needs both chopsticks to eat and always picks up the left chopstick before the right one.
+// A deadlock could potentially occur if all the philosophers reached for the left chopstick at the same time.
+// Using threads and locks, implement a simulation of the dining philosopher’s problem that prevents deadlocks.
+
```

Comment:
A real world implementation of this concept existed at Amazon but we used a cluster_lock to prevent other programs from interfering in one that was making critical changes to a host. I'm not sure of the logic used but it was like a 500+ line java script.